_pages/algorithm-development/create-scm-backed-algo-api.md

```diff
@@ -0,0 +1,143 @@
+
+### Checking your account's SCM authorization status
+
+Before attempting to create your algorithm, it's a good idea to first ensure that the SCM configuration and authorization process has been completed successfully. If the request below returns `{"authorization_status":"unauthorized"}`, you'll need to re-check in the UI to ensure that you authorized the correct configuration (there may be multiple configurations on your cluster for any given SCM provider).
+
+**REST request**
+
+```bash
+$ curl -X POST https://CLUSTER_DOMAIN/v1/scms/SCM_CONFIG_ID/token/status \
+    -H 'Authorization: Simple STD_API_KEY' \
+    -H 'Content-type: application/json'
+```
+
+**REST response**
+
+```json
+{
+  "scm_username": "SCM_USERNAME",
+  "authorization_status": "authorized",
+  "scm_organizations": [
+    {
+      "scm_org_username": "SCM_ORG_USERNAME",
+      "access_level": "MEMBER"
+    }
+  ]
+}
+```
+
+To make sense of this response, suppose that the `SCM_CONFIG_ID` used in the request above was that of a GitHub configuration. The `scm_username` returned would be the GitHub username authorized through the Algorithmia account, and the `scm_org_username` would be the GitHub account of the organization in which the SCM configuration was created by the Algorithmia cluster admin. Also note that in the response, the string `MEMBER` is not a placeholder, but is actually the string that is returned, indicating that the user is a member of the given organization and has corresponding resource access privileges.
+
+### Creating an algorithm
+
+The algorithm request payload shown below has several `UPPERCASE` placeholders that you'll need to replace in your request. In particular, you'll want to make sure that you:
+
+*   Replace the `SCM_CONFIG_ID` value with the one used above, which you've ensured is authorized on the cluster.
+*   Replace the `SCM_USERNAME` value (to specify `repository_owner`) with the value returned from the request above to get the SCM authorization status.
+*   For `repository_name`, you can choose whatever name makes the most sense to you. Often, it will be clearest to name the repository with the same name as the algorithm (as shown in the code below), but you have the option to name it differently if you choose. Note that you'll need to substitute `ENV_ID` with a valid `id` of an algorithm environment on you cluster. You can find an appropriate `id` value by [listing available languages and environments](/developers/algorithm-development/environments#option-b-using-rest-requests-directly).
```

Comment:
```suggestion
*   For `repository_name`, you can choose whatever name makes the most sense to you. Often, it will be clearest to name the repository with the same name as the algorithm (as shown in the code below), but you have the option to name it differently if you choose. Note that you'll need to substitute `ENV_ID` with a valid `id` of an algorithm environment on your cluster. You can find an appropriate `id` value by [listing available languages and environments](/developers/algorithm-development/environments#option-b-using-rest-requests-directly).
```